### PR TITLE
Text expansion shortcut

### PR DESCRIPTION
**Pull request by tangollama**
Based entirely on the work of @eddiesholl in 2016, this is the completed feature for review.

**Changes proposed in this pull request:**
- Added model for text-expansion
- Route for CRUD functions of text expansion
- Navigation and locale changes
- Inclusion of the new component in patients, procedures, visits, labs, imaging, incidents, appointments, and inventory

screen shot 2017-04-18 at 3 10 49 pm

screen shot 2017-04-18 at 3 10 21 pm

cc @HospitalRun/core-maintainers

### CODE REVIEW

File: app/locales/en/translations.js [hunk 1]
```diff
@@ -208,6 +208,17 @@ export default {
       messages: { roleSaved: 'The {{roleName}} role has been saved.' },
       titles: { roleSaved: 'Role Saved' }
     },
+    text_replacements: {
```
Comment (jkleinsc) [inline app/locales/en/translations.js]: @tangollama these need to be camel cased.

**Review by jkleinsc — CHANGES_REQUESTED** [review]
@tangollama looks good overall, but the translations need to be camelCased instead of using underscores.  Can you also merge in the latest from master?

**Comment by tangollama** [comment]
@jkleinsc bump. Changes checked in.

**Review by jkleinsc — APPROVED** [review]

**Comment by sthutchinson** [comment]
**I like this.**   _It would really help with notes_._

**Pull request merged**